commit 2aaf6edd31d6
Author: Kartikaya Gupta <kgupta@mozilla.com>
Date:   Mon Dec 18 13:33:56 2017 -0500

    Bug 1425878 - Don't expose raw pointers to refcounted vsync dispatcher object. r=sotaro
    
    MozReview-Commit-ID: 5ywkZqfY8uZ
    
    --HG--
    extra : rebase_source : 3a887a5765fb0f7c2b5acdc8be94f3ea84199081
---
 widget/nsBaseWidget.cpp | 5 +++--
 widget/nsBaseWidget.h   | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git widget/nsBaseWidget.cpp widget/nsBaseWidget.cpp
index e4ca5e011894..da7b7b259d8d 100644
--- widget/nsBaseWidget.cpp
+++ widget/nsBaseWidget.cpp
@@ -1255,10 +1255,11 @@ void nsBaseWidget::CreateCompositorVsyncDispatcher()
   }
 }
 
-CompositorVsyncDispatcher*
+already_AddRefed<CompositorVsyncDispatcher>
 nsBaseWidget::GetCompositorVsyncDispatcher()
 {
-  return mCompositorVsyncDispatcher;
+  RefPtr<CompositorVsyncDispatcher> dispatcher = mCompositorVsyncDispatcher;
+  return dispatcher.forget();
 }
 
 already_AddRefed<LayerManager>
diff --git widget/nsBaseWidget.h widget/nsBaseWidget.h
index 3cb56f38b6ce..a7b648806bcc 100644
--- widget/nsBaseWidget.h
+++ widget/nsBaseWidget.h
@@ -209,7 +209,7 @@ public:
   // returned.
   void NotifyCompositorSessionLost(mozilla::layers::CompositorSession* aSession);
 
-  mozilla::CompositorVsyncDispatcher* GetCompositorVsyncDispatcher();
+  already_AddRefed<mozilla::CompositorVsyncDispatcher> GetCompositorVsyncDispatcher();
   void            CreateCompositorVsyncDispatcher();
   virtual void            CreateCompositor();
   virtual void            CreateCompositor(int aWidth, int aHeight);
